src/onnx/parse_qlinearunary.cpp

```diff
@@ -0,0 +1,299 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2023 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/common.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/onnx/broadcast_qdq.hpp>
+#include <migraphx/op/pooling.hpp>
+#include <migraphx/instruction.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+void check_qlinearunary_inputs(const op_desc& opd, const std::vector<instruction_ref>& args)
+{
+    if(args.size() < 4)
+        MIGRAPHX_THROW(opd.op_name + ": missing inputs");
+
+    const auto& in_x = args[0];
+
+    auto sh_x   = in_x->get_shape();
+    auto type_x = sh_x.type();
+    if(type_x != migraphx::shape::int8_type and type_x != migraphx::shape::uint8_type)
+        MIGRAPHX_THROW(opd.op_name + ": unsupported input type");
+}
+
+template <class F>
+instruction_ref parse_qlinearunary(const onnx_parser::node_info& info,
```

Comment:
Move this function to a seperate header, similiar to the other helper functions.

Reply:
Also this function should be named `add_qlinearunary` not with `parse_` which is generally reserved for the parser classes.

Reply:
Removed this function because the change you mentioned below makes this solution obsolete.